pkg/hintrunner/zero/zerohint_uint256.go

```diff
@@ -598,3 +598,73 @@ func createUint256SubHinter(resolver hintReferenceResolver) (hinter.Hinter, erro
 
 	return newUint256SubHint(a, b, res), nil
 }
+
+// Uint128Add hint computes the result of the sum of parts of
+// two `uint128` variables(`a` & `b`)  and checks for overflow
+//
+// `newUint128AddHint` takes 3 operanders as arguments
+//   - `a` and `b` are the two `uint128` variables that will be added
+//   - `carry` represent the potential extra bit that needs to be carried
+//     if the res of the sum of `a` and `b` exceeds 2**64 - 1
+
+func newUint128AddHint(a, b, carry hinter.ResOperander) hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "Uint128Add",
+		Op: func(vm *VM.VirtualMachine, _ *hinter.HintRunnerContext) error {
+			//>res = ids.a + ids.b
+			//>ids.carry = 1 if res >= ids.SHIFT else 0
+
+			a, err := hinter.ResolveAsFelt(vm, a)
+			if err != nil {
+				return err
+			}
+
+			b, err := hinter.ResolveAsFelt(vm, b)
+			if err != nil {
+				return err
+			}
+
+			// Calculate `carry` memory value
+			res := new(fp.Element).Add(a, b)
+			var c *fp.Element
+			if utils.FeltLe(&utils.FeltMax128, res) {
+				c = &utils.FeltOne
+			} else {
+				c = &utils.FeltZero
+			}
+
+			cValue := memory.MemoryValueFromFieldElement(c)
+
+			// Save `carry` value in address
+			addrCarry, err := carry.GetAddress(vm)
+			if err != nil {
+				return err
+			}
+
+			err = vm.Memory.WriteToAddress(&addrCarry, &cValue)
```

Comment:
hey, you can directly return `vm.Memory.WriteToAddress(&addrCarry, &cValue)` instead of checking for an error later. If the write was successful, then it will return `nil`.

Reply:
fixed now